services/mailer/mail.go

```diff
@@ -79,24 +81,21 @@ func SendUserMail(language string, u *models.User, tpl base.TplName, code, subje
 	SendAsync(msg)
 }
 
-// Locale represents an interface to translation
-type Locale interface {
-	Language() string
-	Tr(string, ...interface{}) string
-}
-
 // SendActivateAccountMail sends an activation mail to the user (new user registration)
-func SendActivateAccountMail(locale Locale, u *models.User) {
-	SendUserMail(locale.Language(), u, mailAuthActivate, u.GenerateActivateCode(), locale.Tr("mail.activate_account"), "activate account")
+func SendActivateAccountMail(lang string, u *models.User) {
+	locale := translation.NewLocale(lang)
```

Comment:
Every caller of this function is now doing task: `ctx.Locale.Language()` which then immediately gets turned into a Locale again here.

What's the reasoning? Is it so that the mailer has a consistent external interface that does not depend on locale?

Reply:
I'd like to remive the locale interface from mail ...

I'll have a look how it can be dine better